phpunit/class-performance-test.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Server-side performance tests
+ *
+ * @package Gutenberg
+ */
+
+// To run these tests, set the RUN_SLOW_TESTS environment variable to a truthy
+// value.
+if ( getenv( 'RUN_SLOW_TESTS' ) ) {
+	class Performance_Test extends WP_UnitTestCase {
+		function test_parse_large_post() {
+			$start = microtime( true );
+			$start_mem = memory_get_usage();
+
+			$html = file_get_contents(
+				dirname( __FILE__ ) . '/fixtures/long-content.html'
+			);
+			$parser = new Gutenberg_PEG_Parser;
+			$result = $parser->parse( $html );
+
+			$time = microtime( true ) - $start;
+			$mem = memory_get_usage() - $start_mem;
+
+			error_log( '' );
+			error_log( 'Memory used (KB) : ' . round( $mem / 1024 ) );
+			error_log( 'Time (seconds)   : ' . ( round( $time * 100 ) / 100 ) );
+
+			$this->assertLessThanOrEqual(
+				0.3,
```

Comment:
This hard-coded number seems problematic in the long run because machines will get faster and faster that we develop on. What if, and this doesn't seem feasible either, we obtain the CPU speed and use that a multiplier for some threshold value? The other thought that came to mind is to use `register_tick_function()` to count the [ticks](http://php.net/manual/en/control-structures.declare.php#control-structures.declare.ticks) that happen when parsing the large content, but I tried it out and on `master` and on this branch the result is both `11`, probably because I was only declaring ticks in the unit test file and not the parser.